### ISSUE TITLE
salt-call pkg.upgrade name=openssl upgrades all packages

### ISSUE THREAD

**Issue by Ch3LL** [saltstack/salt#34851]
### Description of Issue/Question

When running `salt-call --local pkg.upgrade name=openssl` it upgrades all packages on the OS instead of upgrading only openssl
### Setup
1. (1) minion that requires updates
### Steps to Reproduce Issue
1. 

```
[root@localhost ~]# salt-call --local pkg.upgrade name=openssl
[INFO    ] Executing command ['yum', '--quiet', 'clean', 'expire-cache'] in directory '/root'
[INFO    ] Executing command ['yum', '--quiet', 'check-update'] in directory '/root'
[INFO    ] Executing command ['rpm', '-qa', '--queryformat', '%{NAME}_|-%{EPOCH}_|-%{VERSION}_|-%{RELEASE}_|-%{ARCH}_|-(none)\n'] in directory '/root'
[INFO    ] Executing command ['yum', '--quiet', '-y', 'upgrade'] in directory '/root'
[INFO    ] Executing command ['rpm', '-qa', '--queryformat', '%{NAME}_|-%{EPOCH}_|-%{VERSION}_|-%{RELEASE}_|-%{ARCH}_|-(none)\n'] in directory '/root'
local:
    ----------
    ModemManager-glib:
        ----------
        new:
            1.1.0-8.git20130913.el7
        old:
            1.1.0-6.git20130913.el7
    NetworkManager:
        ----------
        new:
            1:1.0.6-30.el7_2
        old:
            1:0.9.9.1-25.git20140326.4dba720.el7_0
    NetworkManager-adsl:
        ----------
        new:
            1:1.0.6-30.el7_2
        old:
    NetworkManager-bluetooth:
        ----------
        new:
            1:1.0.6-30.el7_2
        old:
    NetworkManager-glib:
        ----------
        new:
            1:1.0.6-30.el7_2
        old:
            1:0.9.9.1-25.git20140326.4dba720.el7_0
    NetworkManager-libnm:
        ----------
        new:
            1:1.0.6-30.el7_2
    btrfs-progs:
        ----------
        new:
            3.19.1-1.el7
        old:
            3.12-4.el7
    bzip2:
        ----------
        new:
            1.0.6-13.el7
        old:
            1.0.6-12.el7
    bzip2-libs:
        ----------
        new:
            1.0.6-13.el7
        old:
            1.0.6-12.el7
    ca-certificates:
        ----------
        new:
            2015.2.6-70.1.el7_2
        old:
            2013.1.95-71.el7
    centos-logos:
        ----------
        new:
            70.0.6-3.el7.centos
        old:
            70.0.6-1.el7.centos
    centos-release:
        ----------
        new:
            7-2.1511.el7.centos.2.10
        old:
            7-0.1406.el7.centos.2.4
    curl:
        ----------
        new:
            7.29.0-25.el7.centos
        old:
            7.29.0-19.el7
    cyrus-sasl-lib:
        ----------
        new:
            2.1.26-20.el7_2
        old:
            2.1.26-17.el7
    dbus:
        ----------
        new:
            1:1.6.12-13.el7
        old:
            1:1.6.12-8.el7
    dbus-libs:
        ----------
        new:
            1:1.6.12-13.el7
        old:
            1:1.6.12-8.el7
    libdb-utils:
        ----------
        new:
            5.3.21-19.el7
        old:
            5.3.21-17.el7_0.1
    libdrm:
        ----------
        new:
            2.4.60-3.el7
        old:
            2.4.50-1.1.el7
    libertas-sd8686-firmware:
        ----------
        new:
        old:
            20140213-0.3.git4164c23.el7
    xz-libs:
        ----------
        new:
            5.1.2-12alpha.el7
        old:
            5.1.2-8alpha.el7
    yum-plugin-fastestmirror:
        ----------
        new:
            1.1.31-34.el7
        old:
            1.1.31-25.el7_0
    zlib:
        ----------
        new:
            1.2.7-15.el7
        old:
            1.2.7-13.el7
```

As shown above it is upgrading a lot more packages then just needed for openssl. 
### Versions Report

```
Salt Version:
           Salt: 2015.8.11

Dependency Versions:
         Jinja2: 2.7.2
       M2Crypto: 0.21.1
           Mako: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.7.0
         Python: 2.7.5 (default, Jun 17 2014, 18:11:42)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
        libgit2: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.7
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: Not Installed
        timelib: Not Installed

System Versions:
           dist: centos 7.0.1406 Core
        machine: x86_64
        release: 3.10.0-123.6.3.el7.x86_64
         system: CentOS Linux 7.0.1406 Core
```

**Comment by Ch3LL** [comment]
also seeing this in 2015.8.9

**Comment by abednarik** [comment]
HI

 I think this is a expected behavior in 2015.8 see [this](https://docs.saltstack.com/en/2015.8/ref/modules/all/salt.modules.yumpkg.html#salt.modules.yumpkg.upgrade) while in 2016.3 this behavior changed to also be able to upgrade single packages  see [this](https://docs.saltstack.com/en/latest/ref/modules/all/salt.modules.yumpkg.html#salt.modules.yumpkg.upgrade)

Cheers

**Comment by rallytime** [comment]
I agree with @abednarik here. I think that is expected in 2015.8, and is fixed when passing with `name=` in 2016.3. I'm going to close this. @Ch3LL Please re-open if you disagree.
